portal-impl/test/integration/com/liferay/portal/service/OrganizationLocalServiceTest.java

```diff
@@ -594,7 +597,135 @@ public void testMoveOrganizationWithSiteToParentOrganizationWithSite()
+			String.valueOf(subOrganization.getOrganizationId()),
+			hits.doc(0).get(Field.ORGANIZATION_ID));
+
+		hits = searchOrganizationsAndUsers(organization, "user1");
+
+		Assert.assertEquals(
+			String.valueOf(user.getUserId()), hits.doc(0).get(Field.USER_ID));
+	}
+
+	@Test
+	public void testSearchOrganizationsAndUsersWhenNoOrganizations()
+		throws Exception {
+
+		Organization organization = OrganizationTestUtil.addOrganization();
+
+		_organizations.add(organization);
+
+		User user = UserTestUtil.addUser("user1", TestPropsValues.getGroupId());
+
+		_users.add(user);
+
+		UserLocalServiceUtil.addOrganizationUsers(
+			organization.getOrganizationId(), new long[] {user.getUserId()});
+
+		Hits hits = searchOrganizationsAndUsers(organization, null);
+
+		Assert.assertEquals(1, hits.getLength());
+
+		Assert.assertEquals(
+			String.valueOf(user.getUserId()), hits.doc(0).get(Field.USER_ID));
+
+		hits = searchOrganizationsAndUsers(organization, "Organization1");
+
+		Assert.assertEquals(0, hits.getLength());
+
+		hits = searchOrganizationsAndUsers(organization, "user1");
+
+		Assert.assertEquals(
+			String.valueOf(user.getUserId()), hits.doc(0).get(Field.USER_ID));
+	}
+
+	@Test
+	public void testSearchOrganizationsAndUsersWhenNoUsers() throws Exception {
+		Organization organization = OrganizationTestUtil.addOrganization();
+
+		Organization subOrganization = OrganizationTestUtil.addOrganization(
+			organization.getOrganizationId(), "Organization1", false);
+
+		_organizations.add(organization);
+		_organizations.add(subOrganization);
+
+		Hits hits = searchOrganizationsAndUsers(organization, null);
+
+		Assert.assertEquals(1, hits.getLength());
+
+		Assert.assertEquals(
+			String.valueOf(subOrganization.getOrganizationId()),
+			hits.doc(0).get(Field.ORGANIZATION_ID));
+
+		hits = searchOrganizationsAndUsers(organization, "Organization1");
+
+		Assert.assertEquals(
+			String.valueOf(subOrganization.getOrganizationId()),
+			hits.doc(0).get(Field.ORGANIZATION_ID));
+
+		hits = searchOrganizationsAndUsers(organization, "user1");
+
+		Assert.assertEquals(0, hits.getLength());
+	}
+
+	@Test
+	public void testSearchOrganizationsAndUsersWithRootOrganization()
+		throws Exception {
+
+		Organization organization = OrganizationTestUtil.addOrganization();
+
+		_organizations.add(organization);
+
+		Hits hits = searchOrganizationsAndUsers(organization, null);
+
+		Assert.assertEquals(0, hits.getLength());
+
+		hits = searchOrganizationsAndUsers(organization, "Organization1");
+
+		Assert.assertEquals(0, hits.getLength());
+
+		hits = searchOrganizationsAndUsers(organization, "user1");
+
+		Assert.assertEquals(0, hits.getLength());
+	}
+
+	protected Hits searchOrganizationsAndUsers(
+			Organization parentOrganization, String keywords)
+		throws Exception {
+
+		return OrganizationLocalServiceUtil.searchOrganizationsAndUsers(
+			parentOrganization.getCompanyId(),
+			parentOrganization.getOrganizationId(), keywords, -1, null, -1, -1,
```

Comment:
I think we should be using the `QueryUtil.ALL_POS` constant instead of `-1` when doing searches like this.